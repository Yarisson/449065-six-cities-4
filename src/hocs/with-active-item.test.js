import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withActiveItem from "./with-active-item.js";

const handleSetActiveItem = function () {

};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActiveItem(MockComponent);

describe(`variants for HOC WithActiveItem`, () => {
  it(`For CityList`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        activeItem={true}
        handleSetActiveItem={handleSetActiveItem}
      />
    ), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With one one count`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        activeItem={[52.390450945458, 4.04309666406198]}
        handleSetActiveItem={handleSetActiveItem}
      />
    ), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
