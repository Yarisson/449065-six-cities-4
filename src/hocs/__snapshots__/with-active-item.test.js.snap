// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variants for HOC WithActiveItem For CityList 1`] = `<div />`;

exports[`variants for HOC WithActiveItem With one one count 1`] = `<div />`;
